pytest_skip/plugin.py

```diff
@@ -124,123 +183,39 @@ def from_config(cls, config: pytest.Config) -> Optional["ShardingConfig"]:
-        return [report_header]
 
     def check_missing_tests(self):
-        missing_test_names = self.test_names - self.seen_test_names
-        if missing_test_names:
-            # If any items remain in `test_names` those tests either don't exist or
-            # have been deselected by another way - warn user
-            if self.select_option == SelectOption.SELECT:
-                message = ("\npytest-skip: Not all selected tests exist "
-                           "(or have been deselected otherwise).\n"
-                           "Missing selected test names:\n  - ")
-            elif self.select_option == SelectOption.DESELECT:
-                message = ("\npytest-skip: Not all deselected tests exist "
-                           "(or have been selected otherwise).\n"
-                           "Missing deselected test names:\n  - ")
-            else:
-                message = ("\npytest-skip: Not all tests to skip exist "
-                           "(or have been not skipped otherwise).\n"
-                           "Missing test names to skip:\n  - ")
-            message += "\n  - ".join(missing_test_names)
-            if self.fail_on_missing:
-                raise UsageError(message)
-            warnings.warn(UserWarning(message))
+        for n, m in self.get_matchers().items():
+            if (missing := set(m.test_names.keys()) - m.seen_test_names):
+                # If any items remain in `test_names` those tests either don't exist or
+                # have been deselected by another way - warn user
+                message = (f"\npytest-skip: Not all tests to {n} exist.\n"
+                           f"Missing test names to {n}:\n  - ")
+                message += "\n  - ".join(missing)
+                if self.fail_on_missing:
+                    raise UsageError(message)
+                warnings.warn(UserWarning(message))
 
     @classmethod
     def from_config(cls, config: pytest.Config) -> Optional["SelectConfig"]:
         fail_on_missing = config.getoption("selectfailonmissing")
-        file_path = None
+        matchers = [None] * 3
         for option in SelectOption:
-            if (option_value := config.getoption(option.value)) is not None:
-                select_option = option
-            else:
+            if (value := config.getoption(option.value)) is None:
                 continue
-            if file_path is None:
-                file_path = option_value
-            else:
-                raise UsageError(
-                    "'--select-from-file', '--deselect-from-file' and '--skip-from-file' cannot be used together."
-                )
-        if file_path is None:
-            return None
-        return SelectConfig(config, select_option, file_path, fail_on_missing)
+            if (matcher := matchers[option.ordinal]) is None:
+                matchers[option.ordinal] = matcher = Matcher()
+            for path in value.split(":"):
```

Comment:
If we have an ability to specify several files, let's add a test for that. Could you also add a test with sharding enabled?